File: Calculator.py
#Start a loop and ask for input, check if input is valid, and ask if the loop should end
B = True
while B:
    #Make boolean to check if operations need to be skipped
    good = True
    #Take in two numbers and the operator as input and print answer
    x = input("Enter your first number:\n")
    #This checks if the first input is a valid number, if not, skip to end
    try:
        type(int(x))
    except:
        print("Not valid input.")
        good = False
        
    if good:    
        operator = input("Do you want to add (+), subtract (-), multiply (*), or divide (/)?\n")
        #This checks if the second input is an operator
        if(operator != '+' and operator != '-' and operator != '/' and operator != '*'):
            print("Not valid input")
            good = False
    
    if good:    
        y = input("Enter your second number:\n")
        #This checks if the third input is a valid number, if not, skip to end
        try:
            type(int(y))
        except:
            print("Not valid input.")
            good = False
        
    if good:    
        #this performs the operation based on the inputs
        if operator == '+': 
            ans = int(x) + int(y)
            
        elif operator == '-': 
            ans = int(x) - int(y)
                
        elif operator == '*':
            ans = int(x) * int(y)
                
        elif operator == '/':
            ans = float(x) / float(y)
            
        try:
            print("Your answer is: " + str(ans))
        except:
            print("Error.") 
        
    restart = input("Do you want to restart? (y/n)\n")
    if restart != 'y':
       B = False

File: Calculator.cpp
#include <iostream>
using namespace std;

int main(){
    //Variables for the calculations
    int x;
    int y;
    char operation;
    //Used to check to break the loop
    char rst;
    //I'm sure this could be handled better, but I used these to hold the two different kinds of answers
    int ans1;
    float ans2;
    while (true){
        cout << "Input the first number: \n";
        cin >> x;
        cout << "Input the operation (+,-,*,/): \n";
        cin >> operation;
        cout << "Input the second number: \n";
        cin >> y;
        cout << "The variables are: " << x << " " << operation << " " << y << "\n";
        switch (operation) {
            case '+' :
                ans1 = x + y;
                cout << "The answer is: " << ans1 << "\n";
                break;
            case '-' :
                ans1 = x - y;
                cout << "The answer is: " << ans1 << "\n";
                break;
            case '*' :
                ans1 = x * y;
                cout << "The answer is: " << ans1 << "\n";
                break;
            case '/' :
                if (y != 0){
                    ans2 = float(x) / float(y);
                    cout << "The answer is: " << ans2 << "\n";
                }else{
                        cout << "Cannot divide by zero. \n";
                }
                break;
            default:
                cout << "Invalid input.";
        }
        cout << "Would you like to restart? (y/n)\n";
        cin >> rst;
        if (rst != 'y')
            break;
    }
    return 0;
}
